### ISSUE TITLE
Support table rows and columns to zoom in and out, and allow users to continue watching the video in the Obsidian table instead of jumping to an external browser to play the video

### ISSUE THREAD

**Issue by zhouxinghong** [decaf-dev/obsidian-dataloom#539]
### Problem

Support table rows and columns to zoom in and out, and allow users to continue watching the video in the Obsidian table instead of jumping to an external browser to play the video

1. The row and column of the table cannot be zoomed in. This cannot zoom in and out for the embedded video screen;
2. There is a problem when playing YouTube videos in Obsidian form. Currently, only videos with the specified link can be played. If I want to continue playing the next video in the obsidian form, it will jump to an external browser instead of continuing to open and play the next video in the obsidian form, which is for users who like to continue watching videos in the obsidian form, It may not be satisfactory! So I hope to strengthen the support for video playback, so that users can continue to watch the video in the obsidian form, instead of jumping to an external browser to play the video. A settings option may be provided to allow the user to enable or disable the ability to continue to the next video in the Obsidian form.
System: Windows 10
Obsidian: 1.3.5
obsidian-notion-like-tables: 6.17
![image](https://github.com/trey-wallis/obsidian-notion-like-tables/assets/54253524/5d0c5365-caca-4bc0-a6fe-331e286c8d86)
Thank you so much for such a wonderful extension!

### Solution

Support table rows and columns to zoom in and out, and allow users to continue watching the video in the Obsidian table instead of jumping to an external browser to play the video

1. The row and column of the table cannot be zoomed in. This cannot zoom in and out for the embedded video screen;
2. There is a problem when playing YouTube videos in Obsidian form. Currently, only videos with the specified link can be played. If I want to continue playing the next video in the obsidian form, it will jump to an external browser instead of continuing to open and play the next video in the obsidian form, which is for users who like to continue watching videos in the obsidian form, It may not be satisfactory! So I hope to strengthen the support for video playback, so that users can continue to watch the video in the obsidian form, instead of jumping to an external browser to play the video. A settings option may be provided to allow the user to enable or disable the ability to continue to the next video in the Obsidian form.
System: Windows 10
Obsidian: 1.3.5
obsidian-notion-like-tables: 6.17
Thank you so much for such a wonderful extension!
### Additional details
Thank you so much for such a wonderful extension!
_No response_

**Comment by trey-wallis** [comment]
@zhouxinghong To "zoom", are you talking about increasing the size of the YouTube video? Try increasing the width size of the column. The embed column by default has an aspect ratio of 16/9 and it will resize to fill the column width. You can also change the aspect ratio in the column settings + adding horizontal/vertical padding if needed.

As for playing the next video in Obsidian, I don't think that this possible at the moment. The YouTube video is playing in an iFrame. I don't control how that is rendered. I only render it.

**Comment by zhouxinghong** [comment]
To "zoom", are you talking about increasing the size of the YouTube video? Try increasing the width size of the column. The embed column by default has an aspect ratio of 16/9 and it will resize to fill the column width. You can also change the aspect ratio in the column settings + adding horizontal/vertical padding if needed.
How to carry out the above setting and operation?

Obsidian already provides the following syntax to render YouTube videos
```html
![](https://www.youtube.com/watch?v=ZSojZyGT8Pg)
```
You can try it!

**Comment by trey-wallis** [comment]
Yep. That is what the table uses to render the YouTube video. I pass it into the Obsidian `MarkdownRenderer.renderMarkdown` function. The table renders the same thing that Obsidian does.

**Comment by zhouxinghong** [comment]
Hope you can perfect the video playback function described above! 👍😜

**Comment by trey-wallis** [comment]
@zhouxinghong To change the column width, hover your mouse near the edge of a column and then drag.

<img width="142" alt="image" src="https://github.com/trey-wallis/obsidian-notion-like-tables/assets/40307803/acb32be3-a968-404b-beb9-2eb1b7036989">


To change the padding or the aspect ratio, open the header edit menu and click `Options`

<img width="182" alt="image" src="https://github.com/trey-wallis/obsidian-notion-like-tables/assets/40307803/7f36d340-192c-4e4e-ad3d-6fe0b7d4c7e4">
<img width="181" alt="image" src="https://github.com/trey-wallis/obsidian-notion-like-tables/assets/40307803/fc135270-ad2a-4a56-94ed-c0f17925a12e">

**Comment by zhouxinghong** [comment]
Thank you for your hard work and effort! I will try the effect later 👍😜🤞

**Comment by zhouxinghong** [comment]
The width of New Column is too narrow. When the title name is relatively long, the input is very inconvenient. It should support widening and shrinking
![2023-06-13_2](https://github.com/trey-wallis/obsidian-notion-like-tables/assets/54253524/d81d9c75-9d7f-488d-b14b-934d0d39df9f)
![image](https://github.com/trey-wallis/obsidian-notion-like-tables/assets/54253524/dada4fb3-32c4-4961-94ab-853a9563f6d7)
Other areas that need improvement will continue to be added later. . . 👍😜

**Comment by zhouxinghong** [comment]
Found a problem:
After using the mouse to drag and zoom in on the video, a vertical scroll bar will appear, and it should be possible to cancel the maximum height limit
